source/slang/slang-check-decl.cpp

```diff
@@ -6916,6 +6945,111 @@ namespace Slang
         }
     }
 
+    void SemanticsDeclBasesVisitor::_searchMembersWithHigherVisibility(StructDecl* structDecl, const DeclVisibility& ctorVisibility, List<VarDeclBase*>& resultMembers)
+    {
+        auto findMembers = [&](StructDecl* structDecl)
+        {
+            for (auto varDeclRef : getMembersOfType<VarDeclBase>(getASTBuilder(), structDecl, MemberFilterStyle::Instance))
+            {
+                auto varDecl = varDeclRef.getDecl();
+                if (getDeclVisibility(varDecl) >= ctorVisibility)
+                {
+                    resultMembers.add(varDecl);
+                }
+            }
+        };
+
+        for (auto inheritanceMember : structDecl->getMembersOfType<InheritanceDecl>())
+        {
+            auto declRefType = as<DeclRefType>(inheritanceMember->base.type);
+            if (!declRefType)
+                continue;
+
+            auto structOfInheritance = as<StructDecl>(declRefType->getDeclRef().getDecl());
+            if (!structOfInheritance)
+                continue;
+
+            findMembers(structOfInheritance);
```

Comment:
For base types, the ctor parameters should be the base type's ctor's parameters, instead of base type's members.

For example, if I have
```
struct Base { int x, int y; __init(int p) {...} }

struct Derived : Base
{
        int z;
}
```

The synthesized ctor for `Derived` should be `__init(int p, int z)` instead of `__init(int x, int y, int z)`.

Reply:
But base type could have more than one ctors, how are we going to handle that?

Reply:
Inheritance is a new issue we haven't clarify in the design doc.

We were aware of that there could be issues we were not able to capture before the implementation. So can we clarify this in the design doc and make this piece more clear, instead of making the notes in github conversation?

Reply:
If base type has user defined explicit ctor, it means we cannot synthesize the ctor the derived type. This is consistent with ordinary fields: if we have a field member T whose type has explicit ctors and the member does not have a default expr, then we cannot synthesize the ctor for the type.

Feel free to update the design doc in this PR.

Reply:
> if we have a field member T whose type has explicit ctors and the member does not have a default expr, then we cannot synthesize the ctor for the type.

I'm not quite sure if this could bring us some problem?

So for your previous example:
```
struct Base { int x, int y; __init(int p) {...} }         // this __init is an explicit ctor

struct Derive: Base
{
    int z;
}
```

in this case, Derive will not have a ctor (neither explicit nor synthesized), so it means that user cannot use initialize list
nor ctor experssion (`Derive d = Derive()`) to initialize it. Is this correct?

Reply:
Correct, this is desired behavior.

Reply:
Got it, thanks for explanation, I will update the design doc.